Makefile

```diff
@@ -1,46 +1,103 @@
-IMAGE_TAG ?= quay.io/rh-jmc-team/container-jfr-operator:0.3.0
-CRDS := containerjfr flightrecorder
+IMAGE_STREAM ?= quay.io/rh-jmc-team/container-jfr-operator
+IMAGE_VERSION ?= 0.3.0
+IMAGE_TAG ?= $(IMAGE_STREAM):$(IMAGE_VERSION)
+
+BUNDLE_STREAM ?= $(IMAGE_STREAM)-bundle
+CSV_VERSION ?= $(IMAGE_VERSION)
+PREV_CSV_VERSION ?= $(CSV_VERSION)
+
+INDEX_STREAM ?= $(IMAGE_STREAM)-index
+INDEX_VERSION ?= 0.0.1
+PREV_INDEX_VERSION ?= $(INDEX_VERSION)
+
+BUILDER ?= podman
 
 .DEFAULT_GOAL := bundle
 
 .PHONY: generate
-generate: k8s openapi
+generate: k8s crds
 
 .PHONY: k8s
 k8s:
 	operator-sdk generate k8s
 
-.PHONY: openapi
-openapi:
-	operator-sdk generate openapi
+.PHONY: crds
+crds:
+	operator-sdk generate crds
+
+.PHONY: csv
+csv: generate
+ifeq ($(CSV_VERSION), $(PREV_CSV_VERSION))
+	operator-sdk generate csv \
+		--csv-config ./deploy/olm-catalog/csv-config.yaml \
+		--csv-version $(CSV_VERSION) \
+		--csv-channel alpha \
+		--default-channel \
+		--update-crds \
+		--operator-name container-jfr-operator-bundle
+else
+	# TODO
+	# from-version should be programatically determined
+	operator-sdk generate csv \
+		--from-version $(PREV_CSV_VERSION) \
+		--csv-config ./deploy/olm-catalog/csv-config.yaml \
+		--csv-version $(CSV_VERSION) \
+		--csv-channel alpha \
+		--default-channel \
+		--update-crds \
+		--operator-name container-jfr-operator-bundle
+endif
 
 .PHONY: image
 image: generate
-	operator-sdk build --image-builder podman $(IMAGE_TAG)
+	operator-sdk build --image-builder $(BUILDER) $(IMAGE_TAG)
 
 .PHONY: bundle
-bundle: image copy-crds
-	operator-courier verify --ui_validate_io bundle
-
-.PHONY: copy-crds
-copy-crds:
-	$(foreach res, $(CRDS), cp -f deploy/crds/rhjmc.redhat.com_$(res)s_crd.yaml bundle/$(res)s.rhjmc.redhat.com.crd.yaml;)
+bundle:
+	operator-sdk bundle create \
+		$(BUNDLE_STREAM):$(CSV_VERSION) \
+		--image-builder $(BUILDER) \
+		--directory ./deploy/olm-catalog/container-jfr-operator-bundle/$(CSV_VERSION) \
+		--default-channel alpha \
+		--channels alpha \
+		--package container-jfr-operator-bundle
+
+.PHONY: validate
+validate: verify-csv
+	operator-sdk bundle validate \
+		--image-builder $(BUILDER) \
+		$(IMAGE_TAG)
+
+.PHONY: verify-csv
+verify-csv:
+	operator-courier verify deploy/olm-catalog/container-jfr-operator-bundle
```

Comment:
Is this functionality still useful? Otherwise we can get rid of our dependency on operator-courier.

Reply:
Maybe not particularly so. It does still work (for now?), and it does validation against the local filesystem whereas the `operator-sdk bundle validate` checks a built image pulled from remote repository. This isn't a huge deal but it definitely seems weird to need to build the bundle image and push it to a private namespace first before being able to validate it.

Reply:
Up to you, but if we keep it, we should probably leave it in the README. Unless we keep `verify-csv`, but remove it as a dependency to `validate`. Then it's just an optional standalone thing.